crates/ol/mempool/src/state.rs

```diff
@@ -0,0 +1,1589 @@
+        // Deserialize transaction from SSZ bytes
+        let tx = OLMempoolTransaction::from_ssz_bytes(&tx_bytes)
+            .map_err(|e| OLMempoolError::Serialization(format!("SSZ decode error: {:?}", e)))?;
+
+        // Compute transaction ID
+        let txid = tx.compute_txid();
+
+        // Add to mempool
+        self.add_transaction(tx).await?;
+
+        Ok(txid)
+    }
+
+    /// Handle get transactions command (returns transactions in priority order).
+    pub(crate) async fn handle_get_transactions(
+        &mut self,
+        limit: usize,
+    ) -> OLMempoolResult<Vec<(OLTxId, OLMempoolTransaction)>> {
+        // Get state accessor (cached or fetched)
+        let state_accessor = self.get_or_fetch_state_accessor().await?;
+
+        let mut result = Vec::new();
+        let mut count = 0;
+
+        // Track expected seq_no per account as we iterate
+        let mut expected_seq_nos: HashMap<AccountId, u64> = HashMap::new();
+
+        // Iterate through transactions in priority order (BTreeMap provides ordered iteration)
+        for txid in self.ordering_index.values() {
+            if count >= limit {
+                break;
+            }
+
+            if let Some(entry) = self.entries.get(txid) {
+                // Apply same filtering as get_all_transactions: check for gaps
+                let should_include = match entry.tx.payload() {
+                    OLMempoolTxPayload::SnarkAccountUpdate(payload) => {
+                        let account = *payload.target();
+                        let seq_no = payload.base_update.operation().seq_no();
+
+                        // Get expected seq_no for this account
+                        let expected_seq_no =
+                            if let Some(&expected) = expected_seq_nos.get(&account) {
+                                expected
+                            } else {
+                                // First time seeing this account, get current on-chain seq_no
+                                if let Ok(Some(account_state)) =
+                                    state_accessor.get_account_state(account)
+                                {
+                                    if let Ok(snark_state) = account_state.as_snark_account() {
+                                        *snark_state.seqno().inner()
+                                    } else {
+                                        continue;
+                                    }
+                                } else {
+                                    continue;
+                                }
+                            };
+
+                        // Only include if no gap
+                        if seq_no == expected_seq_no {
+                            // Update expected seq_no for next transaction from this account
+                            expected_seq_nos.insert(account, seq_no + 1);
+                            true
+                        } else {
+                            false
```

Comment:
right now my understanding is that this cannot happen: we iterate over txs in order, but we are guaranteed to have no gaps by mempool rules. If we actually do get here, this should be an error (at least an error log).